extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1031,6 +1049,52 @@ export class Project implements ISqlProject {
 			}
 		}
 	}
+
+	/**
+	 * Evaluates the value of the property item in the loaded project.
+	 *
+	 * @param propertyName Name of the property item to evaluate.
+	 * @returns Value of the property or `undefined`, if property is missing.
+	 */
+	private evaluateProjectPropertyValue(propertyName: string): string | undefined;
+
+	/**
+	 * Evaluates the value of the property item in the loaded project.
+	 *
+	 * @param propertyName Name of the property item to evaluate.
+	 * @param defaultValue Default value to return, if property is not set.
+	 * @returns Value of the property or `defaultValue`, if property is missing.
+	 */
+	private evaluateProjectPropertyValue(propertyName: string, defaultValue: string): string;
+
+	/**
+	 * Evaluates the value of the property item in the loaded project.
+	 *
+	 * @param propertyName Name of the property item to evaluate.
+	 * @param defaultValue Default value to return, if property is not set.
+	 * @returns Value of the property or `defaultValue`, if property is missing.
+	 */
+	private evaluateProjectPropertyValue(propertyName: string, defaultValue?: string): string | undefined {
+		// TODO: Currently we simply read the value of the first matching element. The code should be updated to:
+		//       1) Narrow it down to items under <PropertyGroup> only
+		//       2) Respect the `Condition` attribute on group and property itself
+		//       3) Evaluate any expressions within the property value
+
+		// Check if property is set in the project
+		const propertyElements = this.projFileXmlDoc.getElementsByTagName(propertyName);
+		if (propertyElements.length === 0) {
```

Comment:
There is a minor change of the behavior, compared to the logic that was hardcoded in the `getProjectTargetVersion`. Previously it was checking that there is only one element found, but now it just checks that there is at least something found and uses the first one. This makes it a little bit closer to reality, as there could be:
```xml
<PropertyGroup>
  <DatabaseCollation>foo</DatabaseCollation>
  <DatabaseCollation Condition="'$(UseCustomCollation)' == 'true'">bar</DatabaseCollation>
</PropertyGroup>
```
Which is a rather common pattern for properties initialization and we should support it, me thinks.